samcli/commands/build/command.py

```diff
@@ -248,22 +249,34 @@ def do_cli(  # pylint: disable=too-many-locals, too-many-statements
 
         try:
             artifacts = builder.build()
-            modified_template = builder.update_template(ctx.template_dict, ctx.original_template_path, artifacts)
 
-            move_template(ctx.original_template_path, ctx.output_template_path, modified_template)
+            stack_output_template_path_by_stack_path = {
+                stack.stack_path: stack.get_output_template_path(ctx.build_dir) for stack in ctx.stacks
+            }
+            for stack in ctx.stacks:
+                modified_template = builder.update_template(
+                    stack.stack_path,
+                    stack.template_dict,
+                    stack.location,
```

Comment:
Just curious, why pass three stack properties individually? and not the whole object? It'll also make documentation bit easier, no?

Reply:
good catch, simplified